lib/model/folder.go

```diff
@@ -450,11 +450,15 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 		}
 	case f.Type == config.FolderTypeReceiveEncrypted:
 		batchAppend = func(fi protocol.FileInfo, _ *db.Snapshot) {
-			// Delete changed items and set zero version vector
-			// such that we get a correct copy again.
-			if err := mtimefs.RemoveAll(fi.Name); err != nil {
-				l.Debugf(`%v Failed to remove changed item "%v": %v`, f.Description(), fi.Name, err)
+			if protocol.IsEncryptedPath(fi.Name) {
+				// Delete changed, encrypted items.
```

Comment:
What happens for stuff that's not under the encrypted path?

Reply:
They get the local flag change. That's what the revert button in the ui is about. Everything that's behind our sentinel gets deleted automatically, everything outside gets the flag such that it shows in UI to be deleted by user action (and gets marked as invalid etc).

Reply:
Are we kosher with stuff being deleted automatically?
We never delete stuff automatically.

Reply:
I know we don't. I thought this might be the one exception: Having files with garbage names and a top-level directory named `a.syncthing-encrypted` should make it highly unlikely that someone puts files in there (as in willfully stupid, not just stupid). At the same time the more I think about it, the less benefit I see in "auto-repairing" stuff (what would we be repairing?). So yeah, maybe we should just mark everything as invalid instead.